src/runtime/memory/memory_manager.cc

```diff
@@ -166,6 +167,12 @@ Allocator* MemoryManager::GetAllocator(Device dev, AllocatorType type) {
   return it->second.at(type).get();
 }
 
+void MemoryManager::Clear() {
+  MemoryManager* m = MemoryManager::Global();
+  std::lock_guard<std::mutex> lock(m->mu_);
+  m->allocators_.clear();
```

Comment:
let us iterate over each allocator and clear each of the freed entries, as oppose to clear the vector, which can cause issues when some of the allocated storage from the allocators get retained and deleted after clear

Reply:
Thanks for pointing it out. Just pushed a new commit, not sure if it addresses in the exactly same way as you mentioned.